src/engine/GroupBy.h

```diff
@@ -190,11 +192,97 @@ class GroupBy : public Operation {
                    const sparqlExpression::EvaluationContext* ctx) {
       if (ValueGetter{}(AD_FWD(value), ctx)) count_++;
     }
-    [[nodiscard]] ValueId calculateResult() const {
+    [[nodiscard]] ValueId calculateResult(const LocalVocab* localVocab) const {
+      (void)localVocab;
       return ValueId::makeFromInt(count_);
     }
   };
 
+  // Data to perform MIN/MAX aggregation using the HashMap optimization.
+  template <valueIdComparators::Comparison Comp>
+  struct ExtremumAggregationData {
+    sparqlExpression::IdOrString currentValue_ = ValueId::makeUndefined();
+    void increment(const sparqlExpression::IdOrString& value,
+                   const sparqlExpression::EvaluationContext* ctx) {
+      if (auto val = std::get_if<ValueId>(&currentValue_);
+          val != nullptr && val->isUndefined()) {
+        currentValue_ = value;
+        return;
+      }
```

Comment:
If I understand the ordinary min/max implementation correctly, the aggregated value is always initialized with the first value, like here. Using `ValueId::max()` here does not work, as the bits for the datatype cannot be interpreted.

Reply:
Okay, so we would need a `maxValueDatatype` that behaves well for that, so you can leave it as is at least for the time being. We can discuss later on how hard this would be.